src/FlashLoanVault.sol

```diff
@@ -0,0 +1,285 @@
+    ///Note: Division of 10000 is for rebasing fee from its integer to percentage form
+    function _flashFee(address token, uint256 amount) internal view returns (uint256) {
+        return amount * fee / 10000;
+    }
+
+    ///@dev Returns the maximum amount of currency available for flash loan
+    ///@param token Address of loan currency
+    ///@return The maximum amount of a token that can be borrowed -> max of DAI deposited
+    ///Note: The maxFlashLoan function MUST return the maximum loan possible for token. If a token is not currently supported maxFlashLoan MUST return 0, instead of reverting
+    function maxFlashLoan(address token) external view returns (uint256) {
+        return supportedTokens[token] ? IERC20(token).balanceOf(address(this)) : 0;
+    }
+
+
+    /*/////////////////////////////////////////////////////////////*/
+    /*                       FRACTIONAL WRAPPER                    */ 
+    /*/////////////////////////////////////////////////////////////*/
+
+    /// @notice Emit event when ERC20 tokens are deposited into Fractionalized Wrapper
+    event Deposit(address indexed caller, address indexed owner, uint256 assets, uint256 shares);
+    
+    /// @notice Emit event when ERC20 tokens are withdrawn from Fractionalized Wrapper
+    event Withdraw(address indexed caller, address indexed receiver, address indexed owner, uint256 assets, uint256 shares);
+
+
+    /// @notice User to deposit underlying tokens for FractionalWrapper tokens
+    /// @dev Mints shares Vault shares to receiver by depositing exactly amount of underlying tokens
+    /// @param assets The amount of underlying tokens to deposit
+    /// @param receiver Address of receiver of Fractional Wrapper shares
+    function deposit(uint256 assets, address receiver) external returns(uint256 shares) {       
+        receiver = msg.sender;
+        shares = convertToShares(assets);
+
+        //transfer DAI from user
+        underlying.safeTransferFrom(receiver, address(this), assets);
+
+        //mint yvDAI to user
+        bool sent = _mint(receiver, shares);
+        require(sent, "Mint failed!"); 
+
+        emit Deposit(msg.sender, receiver, assets, shares);  
+    }
+
+
+    /// @dev Burns shares from owner and sends exactly assets of underlying tokens to receiver; based on the exchange rate.
+    /// @param assets The amount of underlying tokens to withdraw
+    /// @param receiver Address of receiver of underlying tokens - DAI
+    /// @param owner Address of owner of Fractional Wrapper shares - yvDAI
+    function withdraw(uint256 assets, address receiver, address owner) external returns(uint256 shares) {
+        shares = convertToShares(assets);
+        
+        // MUST support a withdraw flow where the shares are burned from owner directly where owner is msg.sender,
+        // OR msg.sender has ERC-20 approval over the shares of owner
+        if(msg.sender != owner){
+            uint allowedShares = _allowance[owner][receiver] ;
```

Comment:
`_allowance` doesn't refer to the token owner and the token receiver, but to the token owner and the function caller. Where you have written `receiver` it should be `msg.sender`.

Reply:
Ah yes, you are absolutely right - the focus should be on owner and function caller; the receiver is irrelevant here. Had to re-read the EIP myself.